modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/ActionRequestProcessor.java

```diff
@@ -165,25 +161,17 @@ private void sendError(RpcContext ctx, RaftErrorCode errorCode, String msg) {
     private void sendError(RpcContext ctx, Status status, Node node) {
         RaftError raftError = status.getRaftError();
 
-        RaftErrorCode raftErrorCode = RaftErrorCode.ILLEGAL_STATE;
-
         PeerId newLeader = null;
 
         if (raftError == RaftError.EPERM) {
             newLeader = node.getLeaderId();
-            PeerId myId = node.getNodeId().getPeerId();
-
-            raftErrorCode = newLeader == null || myId.equals(newLeader) ?
-                RaftErrorCode.NO_LEADER : RaftErrorCode.LEADER_CHANGED;
         }
-        else if (status.getRaftError() == RaftError.ESTATEMACHINE)
-            raftErrorCode = RaftErrorCode.STATE_MACHINE;
 
-        RaftErrorResponseBuilder resp =
-            factory.raftErrorResponse().errorCode(raftErrorCode).errorMessage(status.getErrorMsg());
+        ErrorResponseBuilder resp =
```

Comment:
Is it possible to use org.apache.ignite.raft.jraft.rpc.RpcResponseFactory#newResponse() here?